batchspawner/batchspawner.py

```diff
@@ -353,6 +360,31 @@ def stop(self, now=False):
                              self.job_id, self.current_ip, self.port)
                 )
 
+    # The progress method is only supported on JupyterHub 0.9, which
+    # only supports Python 3.5+.  It requires asyncio coroutines.
+    if sys.version_info >= (3, 5):
+        @async_generator
+        @asyncio.coroutine
```

Comment:
Combining these two things doesn't seem to work on python3.6 at least.  Is there any way to make async_generator work on python 3.5/3.6 without using syntax incompatible with python<3.4?

Reply:
I wonder if using `yield_from_` should be used instead of `yield from yield_` [ref](https://async-generator.readthedocs.io/en/latest/reference.html#yield-from)

Reply:
No luck... I have a feeling it's something about the interaction of `@async_generator` and `@asyncio.coroutine`, but my experience is limited...

This is the actual traceback:
```
    HTTPServerRequest(protocol='http', host='jupyter.triton.aalto.fi', method='GET', uri='/dev/hub/api/users/darstr1/server/progress', version='HTTP/1.1', remote_ip='::ffff:127.0.0.1')
    Traceback (most recent call last):
      File "/share/apps/jupyterhub/dev/miniconda/lib/python3.6/site-packages/tornado/web.py", line 1543, in _execute
        result = yield result
      File "/share/apps/jupyterhub/dev/jupyterhub/jupyterhub/apihandlers/users.py", line 496, in get
        async for event in events:
      File "/share/apps/jupyterhub/dev/miniconda/lib/python3.6/site-packages/async_generator/_impl.py", line 271, in __anext__
        return await self._do_it(self._it.__next__)
      File "/share/apps/jupyterhub/dev/miniconda/lib/python3.6/site-packages/async_generator/_impl.py", line 290, in _do_it
        return await ANextIter(self._it, start_fn, *args)
      File "/share/apps/jupyterhub/dev/miniconda/lib/python3.6/site-packages/async_generator/_impl.py", line 199, in __next__
        return self._invoke(self._it.__next__)
      File "/share/apps/jupyterhub/dev/miniconda/lib/python3.6/site-packages/async_generator/_impl.py", line 209, in _invoke
        result = fn(*args)
      File "/share/apps/jupyterhub/dev/jupyterhub/jupyterhub/utils.py", line 480, in iterate_until
        await yield_(item_future.result())
      File "/share/apps/jupyterhub/dev/miniconda/lib/python3.6/site-packages/async_generator/_impl.py", line 271, in __anext__
        return await self._do_it(self._it.__next__)
      File "/share/apps/jupyterhub/dev/miniconda/lib/python3.6/site-packages/async_generator/_impl.py", line 290, in _do_it
        return await ANextIter(self._it, start_fn, *args)
      File "/share/apps/jupyterhub/dev/miniconda/lib/python3.6/site-packages/async_generator/_impl.py", line 197, in __next__
        return self._invoke(first_fn, *first_args)
      File "/share/apps/jupyterhub/dev/miniconda/lib/python3.6/site-packages/async_generator/_impl.py", line 209, in _invoke
        result = fn(*args)
      File "/share/apps/jupyterhub/dev/jupyterhub/jupyterhub/spawner.py", line 733, in _generate_progress
        await yield_(event)
      File "/share/apps/jupyterhub/dev/miniconda/lib/python3.6/site-packages/async_generator/_util.py", line 14, in __aexit__
        await self._aiter.aclose()
    AttributeError: 'generator' object has no attribute 'aclose'
```

Reply:
Python 3.3 hit end-of-support last year, so I think new releases of everything should drop support for 3.3, if that helps the syntax support issues.

```python
yield from yield_({event})
```

should be the right syntax for 3.4, I think.